converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Extensions/IHttpClientBuilderRegistrationExtensions.cs

```diff
@@ -13,31 +14,33 @@ namespace Microsoft.Health.DicomCast.Core.Extensions
 {
     public static class IHttpClientBuilderRegistrationExtensions
     {
-        public static void AddAuthenticationHandler(this IHttpClientBuilder httpClientBuilder, IServiceCollection services, AuthenticationConfiguration authenticationConfiguration, string credentialProviderName)
+        public static void AddAuthenticationHandler(this IHttpClientBuilder httpClientBuilder, IServiceCollection services, IConfigurationSection authenticationConfigurationSection, string credentialProviderName)
         {
             EnsureArg.IsNotNull(httpClientBuilder, nameof(httpClientBuilder));
             EnsureArg.IsNotNull(services, nameof(services));
-            EnsureArg.IsNotNull(authenticationConfiguration, nameof(authenticationConfiguration));
+            EnsureArg.IsNotNull(authenticationConfigurationSection, nameof(authenticationConfigurationSection));
             EnsureArg.IsNotNullOrWhiteSpace(credentialProviderName, nameof(credentialProviderName));
 
-            if (!authenticationConfiguration.Enabled)
+            var auth = new AuthenticationConfiguration();
+            authenticationConfigurationSection.Bind(auth);
+            if (!auth.Enabled)
             {
                 return;
             }
 
-            switch (authenticationConfiguration.AuthenticationType)
+            switch (auth.AuthenticationType)
```

Comment:
As talked about offline, I think we can push this all off into a separate class loaded via DI such that the configuration resolution binds later, but it's not necessary. It would be sort of how like `ILogger` providers are loaded, or even `IOptions` itself.

Reply:
In theory, you could even create services for every auth scenario, but only read the correct service based on the config when creating an instance of `NamedCredentialProvider`